### ISSUE TITLE
Bug Report: Conditional/Occurrence Rules Report False Negatives

### ISSUE THREAD

**Issue by ww2283** [jdkato/txtlint#1048]
### Check for existing issues

- [x] Completed

## Environment

- **Vale version**: 3.12.0
- **Operating system**: macOS 15.7.1
- **Installation method**: brew install

## Bug Description

Conditional and occurrence extension rules report false negatives when checking content that clearly satisfies the rule criteria. The rules correctly detect patterns in isolated test files but fail when applied through `.vale.ini` configuration sections.

## Expected Behavior

Rules should correctly detect patterns present in the content:
- `RequireDataSource` (conditional rule): Should PASS when content contains "Using our existing dataset"
- `MethodEmphasis` (occurrence rule): Should count 4 keyword matches ("systematic", "method"×2, "analyze")

## Actual Behavior

Rules report false negatives:
- `RequireDataSource`: Reports ERROR claiming pattern is missing, despite being present in content
- `MethodEmphasis`: Reports count=0 or count=1, despite 4 matches being present

### 1. Vale Configuration (`.vale.ini`)

```ini
StylesPath = tests/styles
MinAlertLevel = warning

[*.md]
BasedOnStyles = Base, TestStyle
```

### 2. Conditional Rule (`tests/styles/TestStyle/RequireDataSource.yml`)

```yaml
extends: conditional
message: "Section 1 must acknowledge existing dataset. Use phrasing like 'Using our existing dataset'."
level: error
scope: paragraph
first: '\bSection 1:'
second: 'Using our existing dataset'
```

### 3. Occurrence Rule (`tests/styles/TestStyle/MethodEmphasis.yml`)

```yaml
extends: occurrence
message: "Section 1 should mention 'method', 'analyze', or 'systematic' at least twice (current: %d)"
level: warning
scope: paragraph
ignorecase: true
min: 2
max: 999
tokens:
  - method
  - analyze
  - systematic
```

### 4. Test Content (`test.md`)

```markdown
**Section 1: Identify and analyze patterns in the dataset.**

Using our existing dataset of approximately 5,000 records, this section will apply and refine a systematic method to analyze key patterns and prioritize the most important findings for further investigation. This method integrates multiple data sources to generate a comprehensive list of actionable insights.

**Deliverable:** A ranked list of 30-50 high-priority findings with supporting evidence.
```

## Steps to Reproduce

1. Create directory structure:
   ```
   test-vale-bug/
   ├── .vale.ini
   ├── test.md
   └── tests/styles/TestStyle/
       ├── RequireDataSource.yml
       └── MethodEmphasis.yml
   ```

2. Run Vale:
   ```bash
   vale test.md
   ```

## Actual Output

```
test.md
 3:1  error    Section 1 must acknowledge existing dataset...  TestStyle.RequireDataSource
 3:1  warning  Section 1 should mention... (current: 0)        TestStyle.MethodEmphasis

✖ 1 error, 1 warning and 0 suggestions in 1 file.
```

## Expected Output

Both rules should pass (no errors/warnings) because:
1. Line 3 contains "Using our existing dataset" (satisfies `RequireDataSource`)
2. Paragraph contains "systematic", "method" (×2), "analyze" = 4 matches (satisfies `MethodEmphasis` min: 2)

### Manual Verification

Running `grep` confirms the patterns are present:

```bash
$ grep -n "Using our existing dataset" test.md
3:Using our existing dataset of approximately 5,000 records...

$ grep -oiE '(method|analyze|systematic)' test.md | wc -l
4
```

### Pattern Variations Tested

The bug persists across multiple `.vale.ini` section patterns:
- `[*.md]` (simplest extension pattern) ❌
- `[test*.md]` (filename pattern) ❌
- `[src/*.md]` (path-prefixed pattern) ❌

### Working Workaround

Rules work correctly when testing isolated snippets in different directories, suggesting the issue is with how Vale processes files matched through `.vale.ini` configuration sections when using conditional/occurrence extensions.

## Impact

This bug prevents reliable use of conditional and occurrence rules for automated content validation in documentation/writing workflows. Users cannot trust that passing Vale checks means criteria are actually satisfied.

**Comment by cpkio** [comment]
Instead of creating a new issue I will comment here about a bug in `conditional` rule (false negatives).

I am parsing raw Asciidoc with regular expressions. Vale is latest (3.13.1, Windows, manual installation).

I expect the `conditional` rule to trigger only if `description` is not found, but it triggers every time.

test.adoc
```
= Testing some example title
:navtitle: Example title
:description: Page description
```

rule.yml
```
extends: conditional
message: No `description` attribute found below
level: suggestion
scope: raw
first: '=\s+[A-z,\x20\xA0]+'
second: ':description:'
```

**Comment by jdkato** [comment]
These are config issues (not false negatives):

1. `occurrence` takes a single `token:` regex, not a `tokens:` list. Use:
   ```yaml
   token: 'method|analyze|systematic'
   ```
   (On 3.12.0 the bad `tokens` key was silently ignored → empty token → count 0; current Vale errors on it.)

2. `scope: paragraph` evaluates **each paragraph separately**. Your file is three paragraphs — the bold `**Section 1: …**` line, the body text, and `**Deliverable:** …`. The body paragraph has 4 matches and **passes**; the other two don't. "Section 1" isn't a unit that spans paragraphs.

3. `conditional` means *"the value matched by `first` must be captured by `second`"* (for acronym-style checks, e.g. `WHO` ↔ `… (WHO)`). It can't express "phrase A implies phrase B": `second: 'Using our existing dataset'` has no capture group, so nothing is recorded and every `Section 1:` is flagged — and `first`/`second` are in different paragraphs anyway.

The "works in isolated snippets" behavior is just the single-paragraph case landing everything in one block — it's the content structure, not `.vale.ini` sections.
